src/vision/status_quo/alan_lost_the_world.md

```diff
@@ -0,0 +1,161 @@
+
+Barbara explains that when you borrow a value in a function, the function doesn't gain the lifetime of that borrow. Instead, the borrow comes with it's own lifetime, separate from the function's. The only time a function can have a non-`'static` lifetime is if the function *closes over* some other variable, like so:
+
+```rust
+fn benchmark_sort() -> usize {
+    let mut num_times_called = 0;
+    let test_values = vec![1,3,5,31,2,-13,10,16];
+
+    test_values.sort_by(|a, b| {
+        a.cmp(b)
+        num_times_called += 1;
+    });
+
+    num_times_called
+}
+```
+
+The function passed to `sort_by` has to borrow something not passed into it, namely the `num_times_called` variable. Hence, it has the lifetime of that borrow, not the whole program, because it can't be called anytime - only when `num_times_called` is a valid thing to read or write.
+
+Async functions, it turns out, *close over all their parameters*! They *have to*, because the futures they return don't actually *have* function parameters that the user controls. So they have to copy everything they get passed into, and if that thing is borrowed, then the entire future is borrowed.
+
+Barbara suggests changing all of the async function's parameters to be owned types. Alan asks Grace, who architected this project. Grace recommends holding a reference to the `Plugin` that owns the `World`, and then borrowing it whenver you need the `World`. That ultimately looks like the following:
+
+At this point, Alan is looking at either rewriting a good chunk of the program to not use contexts, or rewriting his own code to sidestep the problem. Alan chooses the latter, by holding a reference to the thing that produces the context and then updating it after the async portion of the task has concluded:
+
+```rust
+async fn load_image(src: String, inside_of: usize, player: Arc<Mutex<Player>>) {
+    let request = make_request(&url);
+    let data = window().unwrap().fetch_with_request(&request).await.unwrap().etc.etc.etc;
+    let image = parse_png(data, context);
+
+    player.lock().unwrap().update(|world| {
+        let new_node_index = world.nodes.len();
+        if let Some(parent) = world.nodes.get(inside_of) {
+            parent.set_child(new_node_index);
+        }
+        world.nodes.push(image.into());
+    });
+}
+```
+
+It works, well enough that Alan is able to finish his changes and PR them into the project. However, Alan wonders if this could be syntactically cleaner, somehow. Right now, async and update code have to be separated - if we need to do something with a `World`, then `await` something else, that requires jumping in and out of this `update` thing. It's a good thing that we only really *have* to be async in these loaders, but it's also a shame that we practically *can't* mix `async` code and `World`s.
+
+## 🤔 Frequently Asked Questions
+
+* **What are the morals of the story?**
```

Comment:
Maybe something like...

"Async functions capture all of their parameters for the entire duration of the function. This allows them to hold borrows of those parameters across await points. However, when the parameter represents the "global environment' in which the future runs, such as the `World` in this story, it may be useful for that parameter not to be captured by the future but rather supplied anew after each await point.